### PR TITLE
Simplified how the Multiselect does array searches.

### PR DESCRIPTION
**Pull request by jimmynotjim**
Based on https://github.com/cfpb/cfgov-refresh/pull/1668#issuecomment-200021403 I simplified the code in the Multiselect that keeps track of the user's selections and updates the dom accordingly.

## Additions

- Adds tests for searchArrayObjects helper.

## Changes

- Uses the native indexOf when available.
- Updates valInArray to searchArrayObjects which return an index when searching for a key:value match instead of the boolean|object confusion.

## Testing

- `gulp test:unit:scripts`
- `gulp build` and use the multiselect (`/blog` is a good spot). It should work as before.

## Review

- @anselmbradford 
- @sebworks 
- @KimberlyMunoz 

## Notes

__Before__

```
=============================== Coverage summary ===============================
Statements   : 26.62% ( 712/2675 )
Branches     : 21.97% ( 192/874 )
Functions    : 19.58% ( 84/429 )
Lines        : 26.72% ( 708/2650 )
================================================================================
```

__After__

```
=============================== Coverage summary ===============================
Statements   : 26.91% ( 720/2676 )
Branches     : 22.43% ( 196/874 )
Functions    : 20.05% ( 86/429 )
Lines        : 27.01% ( 716/2651 )
================================================================================
```

## Todos

-

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: test/unit_tests/modules/util/array-helpers-spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+var expect = chai.expect;
+var jsdom = require( 'mocha-jsdom' );
+var BASE_JS_PATH = '../../../../cfgov/unprocessed/js/';
+var ERROR_MESSAGES = require( BASE_JS_PATH + 'config/error-messages-config' );
+var arrayHelpers = require( BASE_JS_PATH + 'modules/util/array-helpers' );
+var array;
+var index;
+
+describe( 'Array Helpers searchArray', function() {
+  jsdom();
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/array-helpers-spec.js]: Is a DOM needed in this test suite?

File: test/unit_tests/modules/util/array-helpers-spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+var chai = require( 'chai' );
+var expect = chai.expect;
+var jsdom = require( 'mocha-jsdom' );
+var BASE_JS_PATH = '../../../../cfgov/unprocessed/js/';
+var ERROR_MESSAGES = require( BASE_JS_PATH + 'config/error-messages-config' );
+var arrayHelpers = require( BASE_JS_PATH + 'modules/util/array-helpers' );
+var array;
+var index;
+
+describe( 'Array Helpers searchArray', function() {
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/array-helpers-spec.js]: Update function name

File: test/unit_tests/modules/util/array-helpers-spec.js (hunk 1 above)
Comment (jimmynotjim) [inline test/unit_tests/modules/util/array-helpers-spec.js]: Good catch.

**Comment by jimmynotjim** [comment]
@anselmbradford addressed comments

**Comment by anselmbradford** [comment]
:+1:

**Comment by sebworks** [comment]
:+1:

**Pull request merged**